packages/model/addon/-private/model.js

```diff
@@ -19,16 +20,38 @@ import {
   PromiseObject,
   recordDataFor,
   recordIdentifierFor,
-  RootState,
+  relationshipFromMeta,
+  typeForRelationshipMeta,
 } from '@ember-data/store/-private';
 
 import Errors from './errors';
-import {
-  relatedTypesDescriptor,
-  relationshipsByNameDescriptor,
-  relationshipsDescriptor,
-  relationshipsObjectDescriptor,
-} from './system/relationships/ext';
+import { computedMacroWithOptionalParams } from './util';
+
+const RecordMeta = new WeakMap();
+function getRecordMeta(record) {
+  let meta = RecordMeta.get(record);
+  if (meta === undefined) {
+    meta = Object.create(null);
+    RecordMeta.set(record, meta);
+    if (DEBUG) {
+      record.____recordMetaCache = meta;
+    }
+  }
+  return meta;
+}
+
+function getWithDefault(meta, prop, value) {
+  if (meta[prop] === undefined) {
+    meta[prop] = value;
+  }
+  return get(meta, prop);
+}
+
+const retrieveFromCurrentState = computedMacroWithOptionalParams(function retrieveState() {
```

Comment:
We don't pass args to `retrieveFromCurrentState` so we don't need to use this right?

Reply:
Correct.  This was just to reduce some duplication.  We can certainly make it like so if you would like...

```
@computed('currentState')
dirtyType() {
    return this._internalModel.currentState.dirtyType;
}
```

Reply:
We need this to noop in the case of feature flagged paths, right?

Reply:
In order for `isDeletedCP` and similar to work the same

Reply:
isDeletedCP uses its own computed and keeps the flags